### PR TITLE
Refactoring in service of a new CanonicalID type

### PR DESCRIPTION
**Pull request by alexwlchan**
This is some refactoring I spotted while working on https://github.com/wellcomecollection/platform/issues/3659

This should be low-stakes, uncontroversial stuff – just general clean ups and tidies.

### CODE REVIEW

**Review by kenoir — APPROVED** [review]

File: common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/PipelineStorageStream.scala [hunk 1]
```diff
@@ -105,7 +105,7 @@ object PipelineStorageStream extends Logging {
     Flow[(Message, NotificationMessage)]
       .map {
         case (message, notificationMessage) =>
-          Bundle(message, notificationMessage.body, 1)
+          Bundle(message, notificationMessage.body, numberOfItems = 1)
```
Comment (jamesgorrie) [inline common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/PipelineStorageStream.scala:108]: [Bah](https://github.com/scalameta/scalafmt/issues/755)

File: common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/elastic/ElasticIndexerTest.scala [hunk 2]
```diff
@@ -160,6 +164,7 @@ class ElasticIndexerTest
           val future = indexer(documents)
 
           whenReady(future) { result =>
+            println(result)
```
Comment (jamesgorrie) [inline common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/elastic/ElasticIndexerTest.scala:167]: 👀

File: pipeline/matcher/src/test/scala/uk/ac/wellcome/platform/matcher/fixtures/MatcherFixtures.scala [hunk 3]
```diff
@@ -132,8 +132,8 @@ trait MatcherFixtures
     sendNotificationToSQS(queue, body = links.workId)
   }
 
-  def ciHash(str: String): String =
-    DigestUtils.sha256Hex(str)
+  def ciHash(ids: String*): String =
+    DigestUtils.sha256Hex(ids.sorted.mkString("+"))
```
Comment (jamesgorrie) [inline pipeline/matcher/src/test/scala/uk/ac/wellcome/platform/matcher/fixtures/MatcherFixtures.scala:136]: I know this PR is just generalising this, but why do we do this?

File: common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/elastic/ElasticIndexerTest.scala (hunk 2 above)
Comment (alexwlchan) [reply common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/elastic/ElasticIndexerTest.scala:167]: Oops, removed in the follow-up PR: https://github.com/wellcomecollection/catalogue/pull/1504/commits/74ea4df83607ec1a9f0275e6de0718277eed3695

File: pipeline/matcher/src/test/scala/uk/ac/wellcome/platform/matcher/fixtures/MatcherFixtures.scala (hunk 3 above)
Comment (alexwlchan) [reply pipeline/matcher/src/test/scala/uk/ac/wellcome/platform/matcher/fixtures/MatcherFixtures.scala:136]: This is one of the kludges we do for the matcher implementation.

Quick matcher recap: it looks at all the MergeCandidates on Works, and uses those to construct a graph of all the Works. Works in the same graph component can potentially be merged together.

Screenshot 2021-03-23 at 16 05 24

We store these graphs in DynamoDB, with one Dynamo item per Work / graph node.

When you update one Work, you want to find all the other Works in the same component, so we use the "component ID" as an opaque string. That goes in a DynamoDB GSI, so you can do something like `index.findAll(componentId = …)` to find all the Works in that component.

I can't remember why exactly we did it this way, but I think the idea is we want these component IDs to be relatively stable for easier debugging across pipelines.

Does that make sense?

**Pull request merged**